troubleshooting/macbook-troubleshooting.md

```diff
@@ -139,3 +139,182 @@ Please see [this thread o the qubes-devel mailing list][macbook-air-2012-5-1].
+TBV3: in case you reach grub shell, you can 
+~~~
+ls
+~~~
+and also reload config file and change it manualy before booting
+~~~
+configfile /EFI/qubes/grub.cfg
+~~~
+then press "e", edit grub cfg and boot pressing Fn+F10
+
+
+### 5. Fix bootloader
+
+*  Fix grub2 configuration, which uses wrong command for EFI boot
+*  analyzing /mnt/sysimage/var/log/anaconda/program.log I found the faulty commands issues by Anaconda installer
+~~~
+chrooot /mnt/sysimage
+~~~
+* edit /boot/efi/EFI/qubes/xen.cfg file putting the following content
+~~~
+[global]
+default=4.4.14-11.pvops.qubes.x868_64
+
+[4.4.14-11.pvops.qubes.x868_64]
+options=loglvl=all
+kernel=vmlinuz-4.4.14-11.pvops.qubes.x86_64
+ramdisk=initramfs-4.4.14-11.pvops.qubes.x86_64.img
+~~~
+
+*  The main mistake is the efibootmgr, that needs the right commands. Just in case, re-apply all the commands, adapting to your own disk layout (-d /dev/sdxxx -p partition_number)
+~~~
+grep Running /mnt/sysimage/var/log/anaconda/program.log | tail -n 20
+efibootmgr -b 0000 -B
+efibootmgr -c -w -L Qubes -d /dev/sda -p 4 -l \\EFI\\qubes\\xen-4.6.1.efi
+/usr/libexec/mactel-boot-setup
+kernel-install add 4.4.14-11.pvops.qubes.x86_64 /boot/vmlinuz-4.4.14-11.pvops.qubes.x86_64
+systemctl disable qubes-netvm
+reboot
+~~~
+
+At rEFInd screen, choose Boot EFI/qubes/xen-4.6.1.efi
+Everything should now be ok, Qubes OS boots using EFI and you will get the last setup screen
+*  select "Qubes OS", do not change anything and click on "Done"
+*  VMs are created, including NetVM
+
+### 6. Fix pulseaudio, which locks CPU freezing the system often for 20 seconds
+
+My macbook has frequent freezes. Looking at journalctl output I saw that pulseaudio locks CPU for 20 seconds, very often.
+The quickest workaround, is to open a dom0 terminal, as root and edit /etc/pulse/client.conf and adding "autospawn = no"
+Then, as normal user, issue command "pulseaudio --kill"
+
+### 7. Fix system freezes due to Broadcom BCM43602
+
+*  If you experiences a system freeze, during VM setup, force a reboot and press OPTION key. 
+   *  You will reach grub shell
+   ~~~
+   configfile /EFI/qubes/grub.cfg
+   ~~~
+   press Fn+F10 to boot without XEN support
+   *  Once booted, press Fn+CTRL+ALT+F4 to open a shell
+   *  Log into system
+   ~~~
+   sudo su -
+   systemctl disable qubes-netvm
+   ~~~
+   Press Fn+F2 and complete setup
+*  reboot and you finally have your Qubes OS
+*  DO NOT launch sys-net machine
+*  Open its setting and remove wifi adapter from the Selected devices, using Qubes Manager or use the following command line. Get the BFD of the adapter and remove it. On my macbook BFD is 04:00.0 and you will use it later on, also
+~~~
+qvm-pci -l sys-net
+qvm-pci -d sys-net 04:00.0
+~~~
+
+Ok, setup is complete and we are almost done.
+* Open a dom0 terminal
+~~~
+sudo su -
+xl pci-assignable-list
+echo 04:00.0 > /sys/bus/pci/drivers/pciback/permissive
+qvm-start sys-net
+xl pci-attach sys-net 04:00.0
+~~~
```

Comment:
Does the device work after such dynamic attach? Generally dynamic attach of PCI devices is something rather fragile... Maybe you've done something wrong while setting up `qubes-pre-netvm.service`? You should be able to check if permissive mode is enabled by reading this file.
Also, just to be sure: try to enable permissive mode (just like you do), but shutdown sys-net, add the device back to it (using qvm-pci or so), and start again.